src/scenes/Partnerships/PartnershipForm/PartnershipForm.tsx

```diff
@@ -18,73 +18,129 @@ import {
   SecuredField,
   SubmitError,
 } from '../../../components/form';
-import { PartnerField } from '../../../components/form/Lookup';
+import {
+  PartnerField,
+  PartnerLookupItem,
+} from '../../../components/form/Lookup';
 import { Nullable } from '../../../util';
 import { CreatePartnershipFormInput } from '../Create';
 import { EditPartnershipFormInput } from '../Edit';
 import { PartnershipFormFragment } from './PartnershipForm.generated';
 
+type PartnershipformValues = Partial<
+  CreatePartnershipFormInput | EditPartnershipFormInput
+> & {
+  partnership?: {
+    partnerLookupItem?: PartnerLookupItem;
+  };
+};
+
 export type PartnershipFormProps<
-  T extends Partial<CreatePartnershipFormInput | EditPartnershipFormInput>
+  T extends PartnershipformValues
 > = DialogFormProps<T> & {
   partnership?: PartnershipFormFragment;
 };
 
 export const hasManagingType = (types: Nullable<readonly PartnerType[]>) =>
   types?.includes('Managing') ?? false;
 
-export const PartnershipForm = <
-  T extends Partial<CreatePartnershipFormInput | EditPartnershipFormInput>
->({
+export const PartnershipForm = <T extends PartnershipformValues>({
   partnership,
   ...rest
-}: PartnershipFormProps<T>) => (
-  <DialogForm<T> {...rest} fieldsPrefix="partnership">
-    {({ values }) => (
-      <>
-        <SubmitError />
-        {!partnership && <PartnerField name="partnerLookupItem" required />}
-        <SecuredField obj={partnership} name="types">
-          {(props) => (
-            <EnumField
-              multiple
-              label="Types"
-              options={PartnerTypeList}
-              layout="two-column"
-              {...props}
-            />
-          )}
-        </SecuredField>
-        {hasManagingType(values.partnership?.types) ? (
-          <SecuredField obj={partnership} name="financialReportingType">
-            {(props) => (
-              <EnumField
-                label="Financial Reporting Type"
-                options={FinancialReportingTypeList}
-                getLabel={displayFinancialReportingType}
-                {...props}
-              />
-            )}
-          </SecuredField>
-        ) : null}
-        {partnership && (
+}: PartnershipFormProps<T>) => {
+  const decorators = useMemo<Array<Decorator<T>>>(
```

Comment:
Defining the decorators here because it needs the generic `T` to match when passed in to `DialogForm` (which receives the same generic).  Let me know if there's a better way

Reply:
The specific generic type really isn't that important for these decorators (esp since keys are just strings due to dot notation).
You could use `PartnershipFormValues` instead and defined the array outside of the component saving the memo step.
Since we know/enforce that the generic conforms to `PartnershipFormValues` as long as we only need types from `PartnershipFormValues` the generic doesn't add anything. Does that make sense?

Reply:
Done, moved outside of component